pkg/cmd/kafka/topic/update/update.go

```diff
@@ -202,6 +203,14 @@ func runCmd(opts *Options) error {
 				return err
 			}
 		}
+
+		if opts.partitionsStr != "" {
```

Comment:
Partition count is used only in of bellow. 
Best to move it into if scope or closer to where it is used

Reply:
Regarding this and comment above, it aligns with how we handle validations for other flags.